app/models/course_lesson_part.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class CourseLessonPart < ApplicationRecord
+  belongs_to :course_lesson
+
+  has_one :next_lesson_part, class_name: "CourseLessonPart", foreign_key: :previous_lesson_part_id
+  belongs_to :previous_lesson_part, class_name: "CourseLessonPart", inverse_of: :next_lesson_part, optional: true
+
+  validates :title, presence: { message: "Enter a title" }
```

Comment:
Do we want to restrict the number of characters? For example id db you are storing it as a string and if you reach more than 256 characters...it will just do nothing as far as I remember. If we don't want to restrict - changing the field to text will allow stpring more characters. Also we could validate on putting just alphanumeric characters

Reply:
around 7 years ago I was working on a project that had string in the db schema that led to bipolar behaviour and tracking it down - lets say allowed me to fully embrace abusive vocabulary